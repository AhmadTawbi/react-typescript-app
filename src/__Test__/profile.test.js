import renderer from 'react-test-renderer';
import { Button } from "antd";
import profileImage from '../images/profile.jpg';

it('changes the class when hovered', () => {
    const profileCard = { name: 'test', email: 'abc@test.com' };
  const component = renderer.create(
    <div className="profileContainer">
            <div className="profileImage">
                <img src={profileImage} alt="profile" />
            </div>
            <div className="profileName">{profileCard.name}</div>
            <div className="profileEmail">{profileCard.email}</div>
            <div className="viewProfile"><Button onClick={() => {}} type="primary">View Profile</Button></div>
        </div>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
