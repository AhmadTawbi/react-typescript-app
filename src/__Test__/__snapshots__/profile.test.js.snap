// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<div
  className="profileContainer"
>
  <div
    className="profileImage"
  >
    <img
      alt="profile"
      src="profile.jpg"
    />
  </div>
  <div
    className="profileName"
  >
    test
  </div>
  <div
    className="profileEmail"
  >
    abc@test.com
  </div>
  <div
    className="viewProfile"
  >
    <button
      className="ant-btn ant-btn-primary"
      onClick={[Function]}
      type="button"
    >
      <span>
        View Profile
      </span>
    </button>
  </div>
</div>
`;

exports[`changes the class when hovered 2`] = `
<div
  className="profileContainer"
>
  <div
    className="profileImage"
  >
    <img
      alt="profile"
      src="profile.jpg"
    />
  </div>
  <div
    className="profileName"
  >
    test
  </div>
  <div
    className="profileEmail"
  >
    abc@test.com
  </div>
  <div
    className="viewProfile"
  >
    <button
      className="ant-btn ant-btn-primary"
      onClick={[Function]}
      type="button"
    >
      <span>
        View Profile
      </span>
    </button>
  </div>
</div>
`;

exports[`changes the class when hovered 3`] = `
<div
  className="profileContainer"
>
  <div
    className="profileImage"
  >
    <img
      alt="profile"
      src="profile.jpg"
    />
  </div>
  <div
    className="profileName"
  >
    test
  </div>
  <div
    className="profileEmail"
  >
    abc@test.com
  </div>
  <div
    className="viewProfile"
  >
    <button
      className="ant-btn ant-btn-primary"
      onClick={[Function]}
      type="button"
    >
      <span>
        View Profile
      </span>
    </button>
  </div>
</div>
`;
